contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/paginator/PagePaginator.java

```diff
@@ -47,11 +44,10 @@
    * @param pageSizeParam The API Query parameter which specifies how many results per page
    */
   public PagePaginator(Builder builder, int limit, int pageSize, String pageParam, String pageSizeParam) {
-    super(builder, paginationMode.PAGE, pageSize, limit > 0);
+    super(builder, paginationMode.PAGE, pageSize, limit);
     this.limit = limit;
     this.pageParam = pageParam;
     this.pageSizeParam = pageSizeParam;
-    this.paginatedUrls = buildPaginatedURLs();
     currentPage = 1;
```

Comment:
@cgivre While testing this PR today I encountered an API that starts counting pages at 0.